src/js/ble_uuid_util.js

```diff
@@ -36,8 +36,36 @@
 
 module.exports.removeDashes = function(uuid) {
   if (uuid) {
-    uuid = uuid.replace(/-/g, '');
+    var start = 0;
+    var end = 0;
+    var str = '';
+    uuid = String(uuid);
+
+    while (end < uuid.length) {
+      if (uuid.charAt(end) !== '-') end++;
+      else {
+        str += uuid.slice(start, end);
+        start = end = end + 1;
+      }
+    }
+
+    str += uuid.slice(start, end);
   }
 
-  return uuid;
+  return str;
 };
+
+module.exports.refineUuid = function(str, joint) {
```

Comment:
@jeuxdeau What do you mean by refineUuid? is it general term used in uuid? I can't guess what the function `refineUuid` does by the name.